tests/std/tests/Dev11_0316853_find_memchr_optimization/test.cpp

```diff
@@ -475,20 +475,55 @@ int main() {
     { // Test pointers
         const char* s = "xxxyyy";
         const char* arr[]{s, s + 1, s + 1, s + 5, s, s + 4, nullptr};
+        const void* arr_void[]{s, s + 1, s + 1, s + 5, s, s + 4, nullptr};
 
         static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr)), decltype(s + 1)>, "should optimize");
-        static_assert(!_Vector_alg_in_find_is_safe<decltype(begin(arr)), nullptr_t>, "should not optimize");
+        static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr)), char*>, "should optimize");
+        static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr)), void*>, "should optimize");
+        static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr)), const volatile void*>, "should optimize");
+        static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr)), nullptr_t>, "should optimize");
+        static_assert(_Vector_alg_in_find_is_safe<decltype(begin(arr_void)), const char*>, "should optimize");

```

Comment:
The implementation naturally handles it, but I think it would be good to test a few more value types with `arr_void`:

* `char*`
* `const void*`
* `void*`